workitem/field_definition.go

```diff
@@ -197,3 +208,73 @@ func compatibleFields(existing FieldDefinition, new FieldDefinition) bool {
 	}
 	return reflect.DeepEqual(existing.Type, new.Type)
 }
+
+// UnmarshalYAML allows us to have a custom unmarshalling from YAML to a FieldDefinition
+func (f *FieldDefinition) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	type rawValue struct {
+		Label       string  `yaml:"label"`
+		Description *string `yaml:"description"`
+		Icon        *string `yaml:"icon"`
+	}
+
+	type rawType struct {
+		Kind          Kind                       `yaml:"kind"`
+		BaseType      *Kind                      `yaml:"base_type,omitempty"`
+		ComponentType *Kind                      `yaml:"component_type,omitempty"`
+		Values        []map[interface{}]rawValue `yaml:"values,omitempty"`
+	}
+
+	type rawFieldDefinition struct {
+		Required    bool    `yaml:"required"`
+		Label       string  `yaml:"label"`
+		Description string  `yaml:"description"`
+		Type        rawType `yaml:"type"`
+	}
+
+	temp := &rawFieldDefinition{}
+	if err := unmarshal(temp); err != nil {
+		return errs.Wrapf(err, "failed to unmashal YAML to rawFieldDefinition")
+	}
+
+	*f = FieldDefinition{
+		Required:    temp.Required,
+		Label:       temp.Label,
+		Description: temp.Description,
+	}
+
+	switch temp.Type.Kind {
+	case KindList:
+		if temp.Type.ComponentType == nil {
+			return errs.Errorf("list field: '%s' is missing a 'component_type' element: %+v", temp.Label, temp)
+		}
+		f.Type = ListType{
+			SimpleType:    SimpleType{Kind: temp.Type.Kind},
+			ComponentType: SimpleType{Kind: *temp.Type.ComponentType},
+		}
+	case KindEnum:
+		if temp.Type.BaseType == nil {
+			return errs.Errorf("enum field: '%s' is missing a 'base_type' member: %+v", temp.Label, temp)
+		}
+		if len(temp.Type.Values) <= 0 {
+			return errs.Errorf("enum field: '%s' is missing a non-empty 'values' array: %+v", temp.Label, temp)
+		}
+		// TODO(kwk): to get rid of this ugly loop in a loop we need to the enum
+		// type to accept annotated enum values with a description, a label and
+		// an icon
+		vals := []interface{}{}
+		for _, rawVal := range temp.Type.Values {
```

Comment:
No. I don't want to improve it. I want to extend the core to be able to actually handle all the extra information on an enum value.